lib/ansible/plugins/strategy/__init__.py

```diff
@@ -917,7 +917,7 @@ def _evaluate_conditional(h):
             return task.evaluate_conditional(templar, all_vars)
 
         skipped = False
-        msg = ''
+        msg = meta_action
```

Comment:
This only changes non-implicit role_complete tasks which aren't implemented yet. I wonder if that should that be an error until it's implemented? Or at least msg could reflect it's a no-op. But besides role_complete, `msg` is only empty if skipped is true, so the fix looks complete without changing this line.

Reply:
> This only changes non-implicit role_complete tasks which aren't implemented yet. I wonder if that should that be an error until it's implemented? Or at least msg could reflect it's a no-op.

Making it an error doesn't feel right to me since `role_complete` is not documented, one would have to dig through the code to use it. Shouldn't we emit any message for `role_complete` since it is effectively an implementation detail that we now use to indicate that the role has run and users do not have to see it in the output? Or is that useful for debugging?

> But besides role_complete, `msg` is only empty if skipped is true, so the fix looks complete without changing this line.

My thinking was that making `meta_action` a default message is a better default than an empty string. But yeah it is not necessary, I just wanted to be defensive for potential new meta tasks.

Reply:
we should have 2 kinds of meta tasks, the 'implicit' or internally generated should not give user much feedback, the 'explicit' or user generated in playbook, should.